### PR TITLE
add the abstract verification machinery to the musings collection

### PR DESCRIPTION
**Pull request by thomas-fossati**
rendered at https://github.com/veraison/docs/blob/verification-process.md/musings/verification-machine.md

### CODE REVIEW

**Comment by thomas-fossati** [comment]
@setrofim @SimonFrost-Arm @yogeshbdeshpande could you please review?  thanks!

File: musings/verification-machine.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+```
+
+### Crypto Verification
+
+> Input:
+>
+> * Evidence to be verified
+> * Identity endorsements
+
+1. Look up the matching identity endorsement (e.g., a raw public key, a trust anchor, etc.)
```
Comment (setrofim) [inline musings/verification-machine.md:127]:
```suggestion
1. Look up the matching identity endorsement / trust anchor (e.g., a raw public key, an x509 cert, etc.)
```

**Review by setrofim — APPROVED** [review]
LGTM in general, with one proviso:

In our current implementation, the appraisal policy isn't involved at each stage, as described here, but is applied as a final evaluation stage prior to attestation result assembly and signing. 

This allows it to override the outcomes of prior stages, and so can be said to be semantically equivalent to being directly invovled in those stages, as long as the stages can be assumed to be independent of each other (beyond the assumption of successful completion of preceeding stages).

**Review by SimonFrost-Arm — APPROVED** [review]
A useful addition to musings.

File: musings/verification-machine.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+* [Endorsed values decoration](#endorsed-values-decoration)
+* [Attestation result assembly and signing](#attestation-result-assembly-and-signing)
+
+> **Notes**:
+>
+> * The appraisal policy is an input to all the processing stages
+> * All non-final processing stages can output a "failure" signal that short circuits the appraisal
+  
+```mermaid
+flowchart TD
```
Comment (yogeshbdeshpande) [inline musings/verification-machine.md:69]: Nice picture, when it renders, I have a doubt.
Just minor clarifications/improvements, below!

`What is the scope of the Outer yellow boxes, is not clear to me, looking at the diagram??
In the centre if the scope is the Verifier entity, then we should label it to either Verifier or
Verifier actions then it all makes sense. Then in the abstraction Verifier Signing key can be folded in the big Verifier Box..

The color of the other boxes can then be kept other than yellow to indicate components external to the main verification logic.

File: musings/verification-machine.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+    Ev --> v3
+    Ecs1 --> v3
+    v3 --> FAIL
+    v3 --> Dcs
+    v3 --> Ecs2
+    P --> v3
+```
+
+### Attestation Result assembly and signing
+
```
Comment (yogeshbdeshpande) [inline musings/verification-machine.md:214]: In your picture Derived Claims (which is solely based on Endorsed Value addition) is one input to AR, but should we not add:
other inputs like Verifier Added Claims which could be based on
some Verifier policy decisions that augment AR?

File: musings/verification-machine.md (hunk 2 above)
Comment (thomas-fossati) [reply musings/verification-machine.md:69]: > `What is the scope of the Outer yellow boxes, is not clear to me, looking at the diagram??

It just highlights different logical elements.

> In the centre if the scope is the Verifier entity, then we should label it to either Verifier or Verifier actions then it all makes sense.

Sure, good point.

> Then in the abstraction Verifier Signing key can be folded in the big Verifier Box..

I think it's better to keep it outside the verifier box.

File: musings/verification-machine.md (hunk 3 above)
Comment (thomas-fossati) [reply musings/verification-machine.md:214]: Conceptually, appraisal policy is input to all stages.

**Comment by thomas-fossati** [comment]
> LGTM in general, with one proviso:
> 
> In our current implementation, the appraisal policy isn't involved at each stage, as described here, but is applied as a final evaluation stage prior to attestation result assembly and signing.
> 
> This allows it to override the outcomes of prior stages, and so can be said to be semantically equivalent to being directly invovled in those stages, as long as the stages can be assumed to be independent of each other (beyond the assumption of successful completion of preceeding stages).

Correct.  However, I wanted to draw a more generic flow than Veraison's - in fact, I don't think Veraison is ever mentioned in this page.
Also, even in Veraison, the policy step is split between the static part handled by the scheme plugin, and the optional dynamic part handled by the policy handler.  So, it's not that clear-cut either.
I think this is a "reasonable" level of detail, which allows slightly different verifier implementations to be represented.

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM!

**Pull request merged**